### DISCUSSION TITLE
Explaining errors and hover action on one key

### DISCUSSION THREAD

**Discussion by Rumi152** [mrcjkb/rustaceanvim#429] [Q&A] [answered]
Hello,
Is it possible to detect when cursor is hovered on code with error? Aand if not then invoke normal hover action. So for example "K" explains error, or opens hover window if no error is present.

**Comment by mrcjkb** [comment, marked as answer]
Hey 👋 

This plugin doesn't provide an API for that.
It could be possible by looking for diagnostics at the current cursor position and [checking if they have an error code](https://github.com/mrcjkb/rustaceanvim/blob/14a5a4bdf23d1971d1a6386b73b89e57590a5e7d/lua/rustaceanvim/commands/diagnostic.lua#L69).

**Reply by Rumi152** [reply]
Ok, thanks for fast response
